### ISSUE TITLE
ignoredNodes not working as expected (with indent rule)

### ISSUE THREAD

**Issue by avetisk** [eslint/eslint#9724]
**Tell us about your environment**

* **ESLint Version:** 4.13.1
* **Node Version:** 9.2.1
* **npm Version:** using yarn 1.3.2 (npm installed version: 5.6.0)

**What parser (default, Babel-ESLint, etc.) are you using?** 
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

module.exports = {
  env: {
    browser: true,
    es6: true,
    node: true,
    jest: true,
  },
  extends: ['eslint:recommended', 'plugin:react/recommended'],
  parser: 'babel-eslint',
  parserOptions: {
    ecmaVersion: 2017,
    ecmaFeatures: {
      experimentalObjectRestSpread: true,
      jsx: true,
    },
    sourceType: 'module',
  },
  plugins: ['react'],
  rules: {
    indent: ['error', 2, { ignoredNodes: ['ConditionalExpression'] }],
    'linebreak-style': ['error', 'unix'],
    quotes: ['error', 'single', { avoidEscape: true }],
    semi: ['error', 'never'],
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const y = {}

export default ['a','b','c'].reduce(
  (xs, x) =>
    x.includes('&')
      ? x
          .replace(/&/g, xs)
          .replace(/\$\w+/g, z => `.${y[z.slice(1)]}`)
      : `${xs} ${x}`.trim(),
  ''
)
```

```bash
./node_modules/.bin/eslint src/modules/css/test.js
```

**What did you expect to happen?**

As documented on the [indent rule doc page](https://eslint.org/docs/rules/indent#ignorednodes), I was expecting that `eslint` would ignore the indent for the ternary expression.

**What actually happened? Please include the actual, raw output from ESLint.**

I have the following errors:

```
/some/path/src/modules/css/test.js
  7:1  error  Expected indentation of 8 spaces but found 10  indent
  8:1  error  Expected indentation of 8 spaces but found 10  indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

**N.B**: Please note that outside of the lambda, the ignore is working fine.

**Comment by avetisk** [comment]
Well, my bad, actually the error is not on the ternary but the chaining.

`prettier` is indenting the chained methods in a way that breaks the linting.

Now I need to found out how to ignore chaining indent...
